### ISSUE TITLE
For some reasons, I can't seem to query products onSale. 

### ISSUE THREAD

**Issue by danielDozie** [wp-graphql/wp-graphql-woocommerce#625]
**Describe the bug**
I honestly don't know if this is a bug or from my end. Either way, I have tried querying products on sale but I get an empty array. I tried using the default playground here and I actually get results when filtering through products on sale.

**To Reproduce**

`query GetProductsOnSale {products(where: {onSale: true} first: 5){ `
		`edges{`
                       `node{`
                            ` id`
              `onSale`
             ` ... on SimpleProduct{`
               ` price`
                `salePrice`
                `regularPrice`
             ` } } }}}`

**Expected behavior**
I expect to see a return result curated by products onSale. But what I get is an empty array. 
But when I switch the onSale value to false, I get results including products whose onSale value is "TRUE"

**Screenshots**
When onSale value is set true
![Screenshot 2022-05-21 at 13 15 34](https://user-images.githubusercontent.com/31423288/169651086-4d6e9e22-298d-4fb5-896a-a44848762ea6.png)

When onSale value is set to false
![Screenshot 2022-05-21 at 13 17 16](https://user-images.githubusercontent.com/31423288/169651219-ae86487f-3a11-4458-bad8-9c6f75e02019.png)

**Plugin Versions**
- **WooGraphQL Version:** 0.11.0
- **WPGraphQL Version:** 1.8.2
- **WordPress Version:** 5.9.3
- **WooCommerce Version:** 6.5.1

**Comment by scottyzen** [comment]
@danielDozie your query is perfect. I just tested it and it works as expected. The error must be on the WooCommerce end.

**Comment by danielDozie** [comment]
@scottyzen glad you mentioned that. Here are some of the things I have tried so far.

re-installed wpgraphql plugin and  all its extensions.
re-installed woocommerce 
tried disabling a few other plugins in cases of possible conflicts.

But nothing I have tried so far seems to be working for me. And I can't possibly pin-point the issue.


One other work-around I tried for my use-case is, I added a field (foodBounties) using ACF... however, I do not know how to filter the products with this custom field I created with ACF.  Any ideas?
![Screenshot 2022-05-21 at 22 16 34](https://user-images.githubusercontent.com/31423288/169669255-d6272812-4fee-45d1-b450-3e8ce8a3b50e.png)
